Updates error message to be translated in other languages. Issue 685

mayasub opened the pull request:
# Description

This change includes updating the error message boxes to be translatable for 3 different languages. The text was added to data.js and now the code references an object in the file, rather than a hard coded message only in English.

Fixes #685

## Type of change

(Check the ones that apply by placing an "x" instead of the space in the [ ] so it becomes [x])

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

(Note what you have done by placing an "x" instead of the space in the [ ] so it becomes [x]. It is hoped you do all of them.)

- [X] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [X] I have removed text in ( ) from the issue request

## Limitations

This may not be a scalable solution as the data.js file continues to expand but my work follows precedent set by other contributors.

mayasub committed:
Updates error message to be translated in other languages. Issue 685

huss reviewed (changes requested):
Thank you to @mayasub for their first PR and addressing this issue.  It works fine but I would appreciate making the following changes:

- You named the translation key to be permissionError. OED uses periods between words in translation strings. Also, I think it would be better to put it with the csv strings so name it csv.download.size.limit (and in that section alphabetically).
- While the PR did not directly show this, a few lines above your changes in src/client/app/utils/exportData.ts has another string that is not tanslated. It would be named csv.download.size.warning where the current untranslated string is "\<p\>File size will be about ${fileSize.toFixed(2)}MB.\</p\>\<p\>Are you sure you want to download\</p\>'. This is not required for you to do but someone else would need to do it to consider issue #685 to be closed.

Please let me know if anything is not clear or you need any help. Many Thanks.

huss commented:
It has been about 2 weeks weeks since the PR review. I wanted to know if @mayasub plans to make these changes or not. It is okay either way it would be nice to know so the project can continue to resolve this PR.

mayasub commented:
Sorry I don’t have capacity to help in this

On Sun, May 8, 2022 at 6:19 PM Steven Huss-Lederman <
***@***.***> wrote:

> It has been about 2 weeks weeks since the PR review. I wanted to know if
> @mayasub <https://github.com/mayasub> plans to make these changes or not.
> It is okay either way it would be nice to know so the project can continue
> to resolve this PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenEnergyDashboard/OED/pull/757#issuecomment-1120496567>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKXG6HY4VIEFK6D5S4O7FWDVJA4VZANCNFSM5UDG7GLQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
*Maya Subramanian (she/her/hers)*
Computer Science at University of Michigan, Class of 2022
President | Kappa <http://kappathetapi.com> Theta Pi - Professional
Technology Fraternity <http://ktpumich.com>
***@***.*** ***@***.***>* | LinkedIn
<https://www.linkedin.com/in/mayasub/> | (248)-881-9243

huss committed:
update translations

- fixed name of translation key.
- add translation for other export strings.
- removed extra comma at end of line.
- did not translate "ok" since assumed okay in all languages but  should change
sometime.

huss committed:
Merge branch 'development' of https://github.com/OpenEnergyDashboard/OED into pr/mayasub/757

Note fixed issue where TooltipHelpComponentAlternative was still
in a couple of files and changed to TooltipHelpComponent.

huss commented:
I updated the PR for translation key and added a couple more translations to export. I also merged with development to resolve conflict. This is now ready to merge.
